src/main/java/com/solbeg/Main.java

```diff
@@ -107,15 +120,17 @@ public static User getUser(UserProvider userProvider) {
      */
     public static Optional<BigDecimal> retrieveCreditBalance(UserBankAccountProvider userBankAccountProvider) {
 
+        return userBankAccountProvider.getUserBankAccount().get().getCreditBalance();
```

Comment:
![image](https://github.com/MusienkoYuriy97/optional-api-advanced/assets/62200771/f4aef1ae-d47b-4c1f-b8c4-96024b287d98)

Reply:
take a loot at map and flatMap in optional

Reply:
fixed